ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+        wind_condition = WindCondition(
+            wind_speed, wind_dir)
+        if sky_model == 'ASHRAEClearSky':
+            sky_condition = OriginalClearSkyCondition.from_analysis_period(
+                analysis_period, sky_properties[0])
+        elif sky_model == 'ASHRAETau':
+            sky_condition = RevisedClearSkyCondition.from_analysis_period(
+                analysis_period, sky_properties[0], sky_properties[-1])
+        return cls(name, day_type, location, dry_bulb_condition,
+                   humidity_condition, wind_condition, sky_condition)
+
+    @property
+    def ep_style_string(self):
+        """Serialize object to an EnerygPlus SizingPeriod:DesignDay.
+
+        returns:
+            ep_string: A full string representing a SizingPeriod:DesignDay.
+        """
+        # Put together the values in the order that they exist in the ddy file
+        ep_vals = [self.name,
+                   self.sky_condition.month,
+                   self.sky_condition.day_of_month,
+                   self.day_type,
+                   self.dry_bulb_condition.max_dry_bulb,
+                   self.dry_bulb_condition.dry_bulb_range,
+                   self.dry_bulb_condition.modifier_type,
+                   self.dry_bulb_condition.modifier_schedule,
+                   self.humidity_condition.hum_type, '',
+                   self.humidity_condition.schedule, '', '',
+                   self.humidity_condition.wet_bulb_range,
+                   self.humidity_condition.barometric_pressure,
+                   self.wind_condition.wind_speed,
+                   self.wind_condition.wind_direction,
+                   self.wind_condition.rain,
+                   self.wind_condition.snow_on_ground,
+                   self.sky_condition.daylight_savings_indicator,
+                   self.sky_condition.solar_model,
+                   self.sky_condition.beam_shced,
+                   self.sky_condition.diff_sched, '', '', '']
+
+        # assign humidity values based on the type of criteria
+        if self.humidity_condition.hum_type == 'Wetbulb' or \
+                self.humidity_condition.hum_type == 'Dewpoint':
+                    ep_vals[9] = self.humidity_condition.hum_value
+        elif self.humidity_condition.hum_type == 'HumidityRatio':
+            ep_vals[11] = self.humidity_condition.hum_value
+        elif self.humidity_condition.hum_type == 'Enthalpy':
+            ep_vals[12] = self.humidity_condition.hum_value
+
+        # assign sky condition values based on the solar model
+        if self.sky_condition.solar_model == 'ASHRAEClearSky':
+            ep_vals[25] = self.sky_condition.clearness
+        if self.sky_condition.solar_model == 'ASHRAETau':
+            ep_vals[23] = self.sky_condition.tau_b
+            ep_vals[24] = self.sky_condition._tau_d
+            ep_vals.pop()
+
+        # put everything together into one string
+        ep_string = ' SizingPeriod:DesignDay,\n'
```

Comment:
Can we use string formatting to put this together? This is currently really hard to read.

Reply:
Yes. Done!